fuel-relayer/src/pending_blocks.rs

```diff
@@ -0,0 +1,617 @@
+use anyhow::Error;
+use ethers_core::{
+    k256::ecdsa::SigningKey,
+    types::{TransactionRequest, H160, U256},
+};
+use ethers_middleware::{
+    gas_escalator::{Frequency, GasEscalatorMiddleware, GeometricGasPrice},
+    NonceManagerMiddleware, SignerMiddleware,
+};
+use ethers_providers::Middleware;
+use fuel_core_interfaces::{
+    model::{BlockHeight, DaBlockHeight, SealedFuelBlock},
+    relayer::RelayerDb,
+};
+use fuel_tx::Bytes32;
+use std::{cmp::max, collections::VecDeque, sync::Arc};
+
+// use the ethers_signers crate to manage LocalWallet and Signer
+use crate::abi;
+use ethers_signers::{LocalWallet, Signer};
+use tracing::{debug, error, info, warn};
+
+/// Pending Fuel Blocks waiting to be finalized inside client. Until then
+/// there is possibility that they are going to be reverted
+pub struct PendingBlocks {
+    signer: LocalWallet,
+    contract_address: H160,
+    // Pending block commits seen on DA layer and waiting to be finalized
+    pending_block_commits: VecDeque<PendingBlock>,
+    // Highest known chain height, used to check if we are seeing lag between block commits and our fule chain
+    chain_height: BlockHeight,
```

Comment:
It's confusing to keep track of which is da and which is fuel when we sometimes use `da_height` vs `height`, and then use `chain_height` vs `fuel_height` here. Without seeing these two vars side by side, I'm not sure what chain_height means when it's being used.

```suggestion
    best_da_height: BlockHeight,
```